Test simple tags in Time

kidder opened the pull request:
## Proposed changes

Add a test helper for testing simple tags.
Use it to test simple tags in Time directory.
Remove unnecessary name functions in simple Time tags.
Cleanup some documentation of simple Time tags.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x] Code
- [x] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

kidder committed:
Add TestHelpers for DataBox

Add test for simple tags

kidder committed:
Test and cleanup tags related to time stepping

Add test of tags.
Remove unnecessary name functions.
Clean up documentation of some tags.
Use db::tag_name instead of name in InitialValue prefix tag.

nilsdeppe commented on tests/Unit/DataStructures/DataBox/TestHelpers.hpp:
```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#pragma once
```
blank line

nilsdeppe replied:
ping

nilsdeppe commented on tests/Unit/DataStructures/DataBox/TestHelpers.hpp:
```diff
@@ -0,0 +1,38 @@
+                "A type cannot be its own tag.");
+  CHECK(::db::tag_name<Tag>() == expected_name);
+  if (detail::is_name_callable_v<Tag>) {
+    INFO("Do not define name for Tag '" << ::db::tag_name<Tag>() << "',");
```
I'm worried that with using `INFO` here you will collect all the messages for all tags that have a `name()` function. Have you verified that the scoping applies correctly?

kidder replied:
Yes, I got one INFO per error message

nilsdeppe replied:
Cool, cool :P

nilsdeppe commented on tests/Unit/DataStructures/DataBox/TestHelpers.hpp:
```diff
@@ -0,0 +1,38 @@
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace test {
```
why not `TestHelpers`?

kidder replied:
`test` is shorter 😛 I'm fine with `TestHelpers`, but then we should document it...

nilsdeppe replied:
We have been using `TestHelpers` and did tell Will to move something else into a `TestHelpers` namespace :P

nilsvu reviewed (commented):
Other than the namespace change, this LGTM. Note that the new test helper is essentially a protocol conformance test (plus a runtime check for the expected name).

kidder commented:
yeah, I had that in mind when writing it...

kidder committed:
Test and cleanup simple tags in ParallelAlgorithms

Add tests of simple tags.
Remove unnecessary name functions.
Use db::tag_name instead of name in prefix tags.

kidder commented:
@nilsleiffischer please look at as I added a commit
